Cargo.toml

```
@@ -18,4 +18,4 @@ edition = "2021"
 [workspace.dependencies]
 diff = { version = "0.1.12", default_features = false }
 regex = { version = "1", default_features = false }
```

```suggestion
regex = { version = "1.8", default_features = false }
```
regex < 1.8 seems not compatible with regex-syntax 0.7 If i correctly understood  #751

---

That was not my understanding of #751.  Regex 1.8 requires regex_syntax 0.7, but regex_syntax 0.7 should be semver compatible with 0.6, so anything that needed 0.6 (such as regex 1.7) should in theory work with 0.7 as well.

If I take this PR and do:

```
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,5 +22,5 @@ rust-version = "1.64"
 
 [workspace.dependencies]
 diff = { version = "0.1.12", default_features = false }
-regex = { version = "1", default_features = false }
+regex = { version = "=1.7", default_features = false }
 regex-syntax = { version = "0.7", default_features = false }
```

It builds and tests fine for me, which suggests that at least 1.7 works.  Probably we should determine our actual minimum regex version and set Cargo.toml to that, but that could be a separate work.

---

Thank you for testing!
In that case, doesn't it result Cargo.lock includes both `regex-syntax 0.7` and `regex-syntax 0.6`?

---

well, this is not a focus of this PR anyway.

---

Yes, Cargo.lock includes 0.6 and 0.7 in that case, you're right.  I had my semver argument totally wrong above, since 0.6 and 0.7 are of course semver incompatible.  However, having both versions in Cargo.lock is more of a mild-negative affecting compile times than an actual break.

If a user for some reason only has regex 1.7 available, but has syntax 0.7, they should still be able to build lalrpop.  They'll just do extra building to get two versions of regex-syntax, with lalrpop using 0.7 and the older regex versions using 0.6.

---

Yeah, it's possible to have two versions, but the more likely thing is that people will use the latest regex version(which lalrpop itself does) in which case this pr means there is only one version of regex-syntax.

---

> It builds and tests fine for me, which suggests that at least 1.7 works. Probably we should determine our actual minimum regex version and set Cargo.toml to that, but that could be a separate work.

I've previously identified that as Regex 1.3, with #782 being able to catch if that increases on accident.

---

I agree.  I tested regex versions this morning before I saw #782 and came to the same 1.3 conclusion.  Not sure if I'll have time to look over #782 fully today, but I have tested the 1.3 conclusion.